packages/typeorm/src/decorators/kuai-entity.decorator.ts

```diff
@@ -0,0 +1,21 @@
+import { Entity, EntityOptions } from 'typeorm'
+
+export enum Location {
+  Data = '_data',
+  Witness = '_witness',
+}
+
+export const KuaiEntity =
+  (nameSuffix: Location, nameOrOptions?: string | EntityOptions, maybeOptions?: EntityOptions) =>
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  (cls: new (...args: any[]) => unknown): void => {
+    let name = cls.name
+    if (nameOrOptions) {
+      if ('string' === typeof nameOrOptions) {
+        name = nameOrOptions
+      } else if (nameOrOptions.name) {
+        name = nameOrOptions.name
+      }
+    }
+    Entity(name + nameSuffix, maybeOptions)(cls)
```

Comment:
Other fields(except the `name`) in `nameOrOptions` will be discarded if `EntityOptions` is passed as the second parameter.